phoenix-pherf/src/main/java/org/apache/phoenix/pherf/jmx/MonitorManager.java

```diff
@@ -0,0 +1,174 @@
+        MONITOR_DETAILS_LIST = Arrays.asList(MonitorDetails.values());
+    }
+
+    public MonitorManager() throws Exception {
+        this(PherfConstants.MONITOR_FREQUENCY);
+    }
+
+    /**
+     *
+     * @param monitorFrequency Frequency at which monitor stats are written to a log file.
+     * @throws Exception
+     */
+    public MonitorManager(long monitorFrequency) throws Exception {
+        this.monitorFrequency = monitorFrequency;
+        MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
+
+        // Register all the monitors to JMX
+        for (MonitorDetails monitorDetails : MONITOR_DETAILS_LIST) {
+            StandardMBean bean = new StandardMBean(monitorDetails.getMonitor(), Monitor.class);
+            ObjectName monitorThreadStatName = new ObjectName(monitorDetails.toString());
+            try {
+                mbs.registerMBean(bean, monitorThreadStatName);
+            } catch (InstanceAlreadyExistsException e) {
+                mbs.unregisterMBean(monitorThreadStatName);
+                mbs.registerMBean(bean, monitorThreadStatName);
+            }
+        }
+        this.rowCount = 0;
+        this.resultHandler = new CSVResultHandler(PherfConstants.MONITOR_FILE_NAME, ResultFileDetails.CSV);
+    }
+
+    @Override
+    public void run() {
+        try {
+            while (!shouldStop()) {
+                isRunning = true;
+                List rowValues = new ArrayList<String>();
+                synchronized (resultHandler) {
+                    for (MonitorDetails monitorDetails : MONITOR_DETAILS_LIST) {
+                        rowValues.clear();
+                        try {
+                            StandardMBean bean = new StandardMBean(monitorDetails.getMonitor(), Monitor.class);
+
+                            Calendar calendar = new GregorianCalendar();
+                            rowValues.add(monitorDetails);
+
+                            rowValues.add(((Monitor) bean.getImplementation()).getStat());
+                            rowValues.add(DateUtil.DEFAULT_MS_DATE_FORMATTER.format(calendar.getTime()));
+                            Result
+                                    result = new Result(ResultFileDetails.CSV, ResultFileDetails.CSV_MONITOR.getHeader().toString(), rowValues);
+                            resultHandler.write(result);
+                        } catch (Exception e) {
+                            throw new FileLoaderRuntimeException("Could not log monitor result.", e);
+                        }
+                        rowCount++;
+                    }
+                    try {
+                        Thread.sleep(getMonitorFrequency());
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
+                    }
+                }
+            }
+        } finally {
+            try {
+                isRunning = false;
+                if (resultHandler != null) {
```

Comment:
It was added when I was trying to debug a concurrency bug. I found the bug elsewhere, so we could potentially remove this now. I didn't notice any perf drops by keeping it so I just left it alone.